File: src/main/java/com/gorkem/vehicle_inspector/dto/response/AuthResponse.java
package com.gorkem.vehicle_inspector.dto.response;

import com.gorkem.vehicle_inspector.entity.SubscriptionPlan;

public class AuthResponse {

    private final String accessToken;
    private final String tokenType;
    private final Long expiresIn;
    private final Long userId;
    private final String fullName;
    private final String email;
    private final SubscriptionPlan subscriptionPlan;
    private final BusinessAccountResponse businessAccount;

    public AuthResponse(
            String accessToken,
            String tokenType,
            Long expiresIn,
            Long userId,
            String fullName,
            String email,
            SubscriptionPlan subscriptionPlan,
            BusinessAccountResponse businessAccount
    ) {
        this.accessToken = accessToken;
        this.tokenType = tokenType;
        this.expiresIn = expiresIn;
        this.userId = userId;
        this.fullName = fullName;
        this.email = email;
        this.subscriptionPlan =
                subscriptionPlan;
        this.businessAccount = businessAccount;
    }

    public String getAccessToken() {
        return accessToken;
    }

    public String getTokenType() {
        return tokenType;
    }

    public Long getExpiresIn() {
        return expiresIn;
    }

    public Long getUserId() {
        return userId;
    }

    public String getFullName() {
        return fullName;
    }

    public String getEmail() {
        return email;
    }

    public SubscriptionPlan getSubscriptionPlan() {
        return subscriptionPlan;
    }

    public BusinessAccountResponse getBusinessAccount() {
        return businessAccount;
    }
}

File: mobile/lib/features/auth/models/auth_response.dart
import 'business_account.dart';

class AuthResponse {
  const AuthResponse({
    required this.accessToken,
    required this.tokenType,
    required this.expiresIn,
    required this.userId,
    required this.fullName,
    required this.email,
    required this.subscriptionPlan,
    this.businessAccount,
  });

  final String accessToken;
  final String tokenType;
  final int expiresIn;
  final int userId;
  final String fullName;
  final String email;
  final String subscriptionPlan;
  final BusinessAccount? businessAccount;

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      accessToken: json['accessToken'] as String? ?? '',
      tokenType: json['tokenType'] as String? ?? 'Bearer',
      expiresIn: (json['expiresIn'] as num?)?.toInt() ?? 0,
      userId: (json['userId'] as num?)?.toInt() ?? 0,
      fullName: json['fullName'] as String? ?? '',
      email: json['email'] as String? ?? '',
      subscriptionPlan: json['subscriptionPlan'] as String? ?? 'FREE',
      businessAccount: json['businessAccount'] is Map
          ? BusinessAccount.fromJson(
              Map<String, dynamic>.from(json['businessAccount'] as Map),
            )
          : null,
    );
  }
}
